spec/support/macros.rb

```diff
@@ -0,0 +1,11 @@
+def set_current_user(user=nil)
+  session[:user_id] = user ? user.id : Fabricate(:user).id
+end
+
+def sign_in(a_user=nil)
+  user = a_user ? a_user : Fabricate(:user)
```

Comment:
this should be simplified to `user ||= Fabricate(:user)` and 1 line above instead use `user` instead of `a_user` - this is called memoization - if user is set to non-nil then it will skip Fabricating a new one.